adapter.go

```diff
@@ -177,10 +177,7 @@ func NewAdapterByDB(client *mongo.Client, config *AdapterConfig) (persist.BatchA
 }
 
 func (a *adapter) open(clientOption *options.ClientOptions, databaseName string, collectionName string) error {
-	ctx, cancel := context.WithTimeout(context.TODO(), a.timeout)
```

Comment:
I mark this PR drafting for now since I see one removal of the timeout context.
I'm not sure about the level of the effect.

Reply:
OK